### ISSUE TITLE
Support Custom AI Providers and Allow Inputting API Key

### ISSUE THREAD

**Issue by jerryokk** [srcbookdev/srcbook#211]
**Description of the Issue:**
Currently, the application only supports selecting predefined AI providers and does not allow users to input the API key for third-party custom providers. This limitation restricts users from utilizing their preferred AI service providers. Therefore, I suggest adding the following features:

1. Support Custom AI Providers: Allow users to input and select AI providers that are not in the predefined list.
2. Allow Inputting API Key: When adding a custom AI provider, provide an input field for users to enter and save the API key for the third-party provider.

**Expected Behavior:**
1. Add an option in the settings page or configuration file to allow users to add custom AI providers.
2. Provide an input field for users to enter the API key when they add a custom AI provider.
3. Validate the entered API key and ensure the configuration is saved successfully.
4. When utilizing the AI service, the application should correctly call the user-provided custom provider and API key.

**Actual Behavior:**
Currently, the application only supports selecting predefined AI providers and does not provide an option to input the API key.

**Additional Information:**
- Affected Version(s): All
- Screenshot of Current Settings Page (if available):

(Please add a screenshot or relevant information of the current settings page here to help developers better understand the issue.)

**Priority:**
Medium

**Labels:**
enhancement, feature request

**Comment by jerryokk** [comment]
![image](https://github.com/user-attachments/assets/d1652b58-e4cc-40b1-837e-b59aec22e5ab)

**Comment by jerryokk** [comment]
> Hey @jerryokk, thanks for raising the issue. Indeed we have no way of using 3rd party providers that aren't local yet.
> 
> Is there a specific provider/model combination that you were hoping to use?

yes, I want to use a third-party cloud service compatible with OpenAI's API.

**Comment by nichochar** [comment]
This is availabe in 0.0.16
